roachtest: remove cloud from registry and ClusterSpec

RaduBerinde opened the pull request:
This change is the last step in removing runtime state from the test
registry and the cluster spec. The cloud is no longer accessible at
test registration time.

Fixes #104029
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/113505)
<!-- Reviewable:end -->

renatolabs reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

Minor comments, nice work!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/113505#-:-NiLp-WV6rDtgiqfGANb:b-8k5twv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde and @srosenberg)

___
*[`pkg/cmd/roachtest/spec/cluster_spec.go` line 252 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/113505#-NiLo8utECy7cDWVtvrK:-NiLo8utECy7cDWVtvrL:b-a77vku) ([raw file](https://github.com/cockroachdb/cockroach/blob/4c2d872ed585ef220c7b3378b3b4cef63e6ee245/pkg/cmd/roachtest/spec/cluster_spec.go#L252)):*
> ```Go
> 	//  - the existing cluster's arch, if we are reusing a cluster;
> 	//  - otherwise, it is the test'
> 	// it; it is not guaranteed
> ```

Documentation is off here.
___
*[`pkg/cmd/roachtest/spec/cluster_spec.go` line 348 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/113505#-NiLoW9s2Rsi-su4cGji:-NiLoW9s2Rsi-su4cGjj:bmw8q2d) ([raw file](https://github.com/cockroachdb/cockroach/blob/4c2d872ed585ef220c7b3378b3b4cef63e6ee245/pkg/cmd/roachtest/spec/cluster_spec.go#L348)):*
> ```Go
> 			// based on the cloud and CPU count.
> 			var err error
> 			switch params.Cloud {
> ```

Nit: we're switching on `cloud` directly above; might make sense to do this here too to lower cognitive load.


<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/113505#-:-NiLv33L-VBLMvxf_z_U:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @renatolabs and @srosenberg)

___
*[`pkg/cmd/roachtest/spec/cluster_spec.go` line 252 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/113505#-NiLo8utECy7cDWVtvrK:-NiLusJ95nxYpk-qN_e3:buzozvq) ([raw file](https://github.com/cockroachdb/cockroach/blob/4c2d872ed585ef220c7b3378b3b4cef63e6ee245/pkg/cmd/roachtest/spec/cluster_spec.go#L252)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Documentation is off here.
</blockquote></details>

Done. Not sure where the change came from, reverted.


<!-- Sent from Reviewable.io -->

RaduBerinde committed:
roachtest: remove cloud from registry and ClusterSpec

This change is the last step in removing runtime state from the test
registry and the cluster spec. The cloud is no longer accessible at
test registration time.

Fixes #104029
Release note: None

RaduBerinde commented:
Rebased and started test run: https://teamcity.cockroachdb.com/viewQueued.html?itemId=12554668

RaduBerinde commented:
bors r+
